src/main/java/Serf.java

```diff
@@ -0,0 +1,100 @@
+                receivedMessage = requestedMessage.nextLine();
+                continue;
+            } else if (receivedMessage.contains("mark") && !receivedMessage.contains("unmark")) { // user keys in mark
+                String number = "";
+                for (int j = 0; j < receivedMessage.length(); j += 1) { // reads number from input and store it in String number
+                    if (Character.isDigit(receivedMessage.charAt(j))) {
+                        number += receivedMessage.charAt(j);
+                    }
+                }
+                if (number.isEmpty()) {
+                    System.out.println(FIVEWHITESPACES + "Error");
+                    continue;
+                } else {
+                    taskList[Integer.parseInt(number) - 1].markAsDone();
+                }
+                System.out.println(VERTICALLINES);
+                System.out.println(FIVEWHITESPACES + "Acknowledged sire, I've marked this task as complete");
+                System.out.println(FIVEWHITESPACES
+                        + "  "
+                        + "["
+                        + taskList[Integer.parseInt(number) - 1].getStatusIcon() + "] "
+                        + taskList[Integer.parseInt(number) - 1].getDescription());
+                System.out.println(VERTICALLINES);
+
+            } else if (receivedMessage.contains("unmark")) { // user keys in unmark
+                String number = "";
+                for (int j = 0; j < receivedMessage.length(); j += 1) { // reads number from input and store it in String number
+                    if (Character.isDigit(receivedMessage.charAt(j))) {
+                        number += receivedMessage.charAt(j);
+                    }
+                }
+                if (number.isEmpty()) {
+                    System.out.println(FIVEWHITESPACES + "Error");
+                    continue;
+                } else {
+                    taskList[Integer.parseInt(number) - 1].markAsUndone();
+                }
+                System.out.println(VERTICALLINES);
+                System.out.println(FIVEWHITESPACES + "Acknowledged sire, I've marked this task as incomplete");
+                System.out.println(FIVEWHITESPACES
+                        + "  "
+                        + "["
+                        + taskList[Integer.parseInt(number) - 1].getStatusIcon() + "] "
+                        + taskList[Integer.parseInt(number) - 1].getDescription());
+                System.out.println(VERTICALLINES);
+            } else {  // add new task to taskList
+                taskList[taskCounter] = new Task(receivedMessage);
+                taskCounter += 1;
+                System.out.println(VERTICALLINES);
+                System.out.println(FIVEWHITESPACES + "added: " + receivedMessage);
+                System.out.println(VERTICALLINES);
+            }
+            receivedMessage = requestedMessage.nextLine();
+        }
+        System.out.println(FAREWELLMESSAGE);
```

Comment:
The code structure is logical. Good that you added comments to increase code understanding as well.